fpsd/features.py

```diff
@@ -0,0 +1,768 @@
+
+        return ["features.burst_length_aggregates",
+                "features.burst_binned_lengths",
+                "features.burst_lengths"]
+
+    def _get_column_tables_of_table(self, schema_name, table_name):
+        query = ("SELECT column_name FROM information_schema.columns "
+                 "WHERE table_schema='{}' "
+                 "AND table_name='{}'").format(schema_name,
+                                               table_name)
+        result = self.engine.execute(query)
+        colnames = []
+        for row in result:
+            colnames.append(row[0])
+        return colnames
+
+    def create_master_feature_view(self, feature_table_names):
+        """This generates a view of all the feature tables at
+        features.frontpage_features in the following form:
+
+        exampleid  | feature_1 | feature_2 ...    | feature_n
+        (integer)  | (bigint)  | (bigint)  ...    | (bigint)
+        --------------------------------------------------------
+        9          | 1         | 2         ...    | 34
+        10         | 2         | 2         ...    | 21
+        11         | 1         | 3         ...    | 36
+
+        A single row represents a single example along with the features
+        for that example. The purpose of the view is to provide a simple
+        way to select train and test sets.
+
+        Args:
+            feature_table_names [list of strings]: list of tables that
+            contain features that we would like to put into the view
+        """
+
+        master_features = {}
+
+        for schema_and_table in feature_table_names:
+            schema_name = schema_and_table.split('.')[0]
+            table_name = schema_and_table.split('.')[1]
+            table_columns = self._get_column_tables_of_table(schema_name,
+                                                             table_name)
+            table_columns.remove("exampleid")
+            master_features.update({schema_and_table: table_columns})
+
+        columns_to_select = "foo.exampleid"
+        full_join_query = ("FROM ( (SELECT exampleid FROM      " 
+                           "features.undefended_frontpage_examples) foo ")
+
+        for table_num, table_name in enumerate(list(master_features.keys())):
+            prefix = 't{}.'.format(table_num)
+            prefixed_columns = [prefix + s for s in master_features[table_name]]
+            columns_to_select = columns_to_select + ', ' + ', '.join(prefixed_columns)
+            join_query = ("LEFT OUTER JOIN {name} t{num} "
+                          "ON foo.exampleid = t{num}.exampleid ").format(name=table_name,
+                                                                         num=table_num)
+            full_join_query = full_join_query + join_query
+
+        drop_view = "DROP VIEW IF EXISTS features.frontpage_features; "
+        self.engine.execute(drop_view)
+
+        create_new_view = ("CREATE VIEW features.frontpage_features "
```

Comment:
Maybe I’m not understanding, if you’re asking me to explicitly write out this query then I want to avoid doing that such that this function can take an arbitrary list of tables in the future and construct a master feature view. Also the feature view can contain nulls, there is an interpolation step in the machine learning pipeline to handle these missing values